File: lib/models/transaction.dart
class Transaction {
  final int id;
  final int studentId;
  final String title;
  final double amount;
  final String transactionType;
  final String status;
  final DateTime? transactionDate;

  Transaction({
    required this.id,
    required this.studentId,
    required this.title,
    required this.amount,
    required this.transactionType,
    required this.status,
    this.transactionDate,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'] ?? 0,
      studentId: json['studentId'] ?? 0,
      title: json['title'] ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      transactionType: json['transactionType'] ?? '',
      status: json['status'] ?? '',
      transactionDate: json['transactionDate'] != null
          ? DateTime.tryParse(json['transactionDate'])
          : null,
    );
  }
}

File: BackEnd/MyFSchool_Backend/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyFSchool_Backend.Models;

public class Transaction
{
    [Key]
    public int Id { get; set; }

    public int StudentId { get; set; }

    [Required]
    [MaxLength(150)]
    public string Title { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    [Required]
    [MaxLength(20)]
    public string TransactionType { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    public string Status { get; set; } = string.Empty;

    [Required]
    public DateTime TransactionDate { get; set; }

    // Navigation properties
    [ForeignKey("StudentId")]
    public User? Student { get; set; }
}
